pyvrp/cpp/Solution.cpp

```diff
@@ -256,7 +368,11 @@ std::ostream &operator<<(std::ostream &out, Solution const &sol)
     auto const &routes = sol.getRoutes();
 
     for (size_t idx = 0; idx != routes.size(); ++idx)
-        out << "Route #" << idx + 1 << ": " << routes[idx] << '\n';
+    {
+        out << "Route #" << idx + 1;
+        out << " (type " << routes[idx].vehicleType() << ")";
+        out << ": " << routes[idx] << '\n';
```

Comment:
Ideally yes, but I think to be valid in the VRPLib format we should print the type before the `:` whereas the`route` is only printed after the `:`. Letting `Route` print the complete line is also not possible since it doesn't know about the idx. I am open to suggestions.

Related to the question _if_ we want this, I'm not sure, some options here, please pick your preference:
* Always print vehicle type (current).
* Don't print the vehicle type at all. I think this skips relevant information.
* Only print vehicle type if we have more than one vehicle type in the solution. This requires extra for loop to check this before printing, and if only one type is used whereas more are available, it is not printed which type is used.
* Only print vehicle type if we have more than one vehicle type in the instance. This requires access to problemData which we don't have and can't pass since it's a parameterless function.
* Only print the vehicle type if it is `!= 0`, which means we will assume a 'default' vehicle type and keep unchanged behavior for the homogeneous case, but it looks somewhat weird for heterogeneous cases.
* Open to other suggestions?

Reply:
Until we can resolve this in a satisfactory way, I think we should revert to "Don't print the vehicle type at all". Right now we're guessing at the appropriate change, which I'd rather not do until we can get it right.

One thing that's for example important to me is that we preserve VRPLIB format in the case of homogeneous fleet. The changes proposed here do not do this, but e.g. the `!= 0` check would. Let's postpone the discussion until we know what to do, and stick to what we had until then.